integration-cli/docker_api_exec_test.go

```diff
@@ -27,3 +29,27 @@ func (s *DockerSuite) TestExecApiCreateNoCmd(c *check.C) {
 		c.Fatalf("Expected message when creating exec command with no Cmd specified")
 	}
 }
+
+func (s *DockerSuite) TestExecApiCreateNoValidContentType(c *check.C) {
+	name := "exec_test"
+	runCmd := exec.Command(dockerBinary, "run", "-d", "-t", "--name", name, "busybox", "/bin/sh")
+	if out, _, err := runCommandWithOutput(runCmd); err != nil {
+		c.Fatal(out, err)
+	}
+
+	jsonData := bytes.NewBuffer(nil)
+	if err := json.NewEncoder(jsonData).Encode(map[string]interface{}{"Cmd": nil}); err != nil {
+		c.Fatalf("Can not encode data to json %s", err)
+	}
+
+	res, body, err := sockRequestRaw("POST", fmt.Sprintf("/containers/%s/exec", name), jsonData, "text/plain")
+	status := res.StatusCode
+
+	c.Assert(status, check.Equals, http.StatusInternalServerError)
+	c.Assert(err, check.IsNil)
```

Comment:
check error before checking status, also for status just use `res.StatusCode` directly :)